CMakeLists.txt

```diff
@@ -572,26 +572,30 @@ if(WIN32)
   add_compile_options(/Zl) # omit default library name in .OBJ
 endif(WIN32)
 
+#--------------------------------
+# Definition directives
+#  - all clr specific compile definitions should be included in this file
+#  - all clr specific feature variable should also be added in this file
+#----------------------------------
+include(clrdefinitions.cmake)
```

Comment:
Why does it need to be built before the CoreCLR PAL? Looks like there is some problem elsewhere. I don't see a reason for this ordering.  The defined symbols, compiler setting etc should be the same whether you build it before or after the CoreCLR PAL.